script.js

```diff
@@ -1,70 +1,96 @@
 (function(global) {
 
-  var requestData = $.getJSON('data.json').done(function(products) {
-    var requestProductTemplate = $.get('product-template.html').done(function(template) {
+  var _template = function(str, param) {
```

Comment:
Template function is following standard convention of templating engine, which takes html string and data as its parameter.